src/runtime/EncodedValue.h

```diff
@@ -38,18 +38,23 @@ COMPILE_ASSERT(sizeof(EncodedValueData) == 4, "");
 COMPILE_ASSERT(sizeof(EncodedValueData) == 8, "");
 #endif
 
-#define ENCODED_VALUE_DOUBLE_TAG_IN_DATA 1
-
+#pragma pack(push, 1)
+// DoubleInEncodedValue stores its tag in `this + sizeof(size_t)`
+// the location is same with PointerValues
 class DoubleInEncodedValue : public gc {
```

Comment:
Actually, `DoubleInEncodedValue` has no gc reference in itself.
So, how about overloading `new` operation with `GC_MALLOC_ATOMIC`?

Reply:
It is already allocated by GC_MALLOC_ATOMIC

Reply:
I see